src/fsharp/MethodOverrides.fs

```diff
@@ -420,7 +459,116 @@ module DispatchSlotChecking =
                     | meth :: _ when meth.IsFinal -> errorR(Error(FSComp.SR.tcCannotOverrideSealedMethod((sprintf "%s::%s" (meth.ApparentEnclosingType.ToString()) (meth.LogicalName))), m))
                     | _ -> ()
 
+    /// Get default slot flags for the given method.
+    let inline GetDefaultDispatchSlotFlags (minfo: MethInfo) =
+        (if minfo.IsAbstract then RequiredSlotFlags.None else RequiredSlotFlags.Optional) |||
+        (if minfo.IsDefaultInterfaceMethod then RequiredSlotFlags.DefaultInterfaceImplementation else RequiredSlotFlags.None)
 
+    /// Get a collection of interface methods that are overrider (aka. override by) slots for the given method from the given types.
+    let GetInterfaceOverriderMethods (infoReader: InfoReader) m (minfo: MethInfo) tys =
+        let g = infoReader.g
+        let amap = infoReader.amap
+        let mty = minfo.ApparentEnclosingType
+
+        if minfo.IsInterfaceMethod then
+            tys
+            |> List.choose (fun ty ->
+                if isInterfaceTy g ty && not (typeEquiv g mty ty) && TypeFeasiblySubsumesType 0 g amap m mty CanCoerce ty then
+                    // We are only looking for IL overriders because F# does not support creating your own default interface methods.
+                    match TryFindILIntrinisicOverriderMethInfo infoReader m AccessibleFromSomewhere ty minfo with
+                    | Some minfo2 when not minfo2.IsNewSlot -> Some minfo2
+                    | _ -> None
+                else
+                    None)
+        else
+            []
+
+    /// Get a collection of interface methods that are the top most overrider methods in the hierarchy for the given method from the required type infos.
+    let GetTopMostHierarchicalInterfaceOverriderMethods (infoReader: InfoReader) m reqdTyInfos (minfo: MethInfo) =
```

Comment:
This is really not that optimized. For every method on an interface, loop through all interfaces that a class would implement, determine whether the method's enclosing type subsumes each interface type and find an override. There might need to be a better way.

Reply:
I think we can determine a lot of this once and not have to do it for every method.